### PR TITLE
sidecar: copy service port while copying matching ports to service

### PR DESCRIPTION
**Pull request by ramaraochavali**
The current implementation while trimming sidecar ports for non matching listener ports is copying port from listener. But EDS code is dependent on port name https://github.com/istio/istio/blob/master/pilot/pkg/proxy/envoy/v2/eds.go#L903. So no endpoints are returned for scoped services. This PR corrects that by copying the service port.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@rshriram Can you PTAL - minor change?

File: pilot/pkg/model/sidecar_test.go [hunk 1]
```diff
@@ -218,6 +226,24 @@ var (
...
+		ConfigMeta: ConfigMeta{
+			Name: "different-port-name",
+		},
+		Spec: &networking.Sidecar{
+			Egress: []*networking.IstioEgressListener{
+				{
+					Port: &networking.Port{
+						Number:   7443,
+						Protocol: "GRPC",
+						Name:     "listener-grpc-tls",
```
Comment (howardjohn) [inline pilot/pkg/model/sidecar_test.go]: this seems weird, shouldn't the name have to match here too?

**Comment by ramaraochavali** [comment]
It does n’t  have to. That is bug this PR fixes. Listener port can be any
thing - it matches by number

**Comment by howardjohn** [comment]
Isn't it weird though that for the Sidecar we match by number but then in EDS we match by name?

**Comment by ramaraochavali** [comment]
In Eds we are matching service port name and endpoint port name which is
fine IMO. Previously here when I cloned service I should have cloned from
service- actually that code incorrectly modified the port of service, which
is corrected here. I think egress listener port name need not have to match
all service portnames - because each service can name the same port
differently I guess.

**Comment by ramaraochavali** [comment]
If you think we should change to only look at port number I am fine
with that  - but not sure of its other implications. Shriram, WDYT?

**Comment by howardjohn** [comment]
I think this is probably find actually, the whole port name matching is
sort of confusing already.

Will let Shriram take a look if he is interested then approve if not

**Comment by ramaraochavali** [comment]
@howardjohn  Cool. Thanks

**Comment by ramaraochavali** [comment]
@howardjohn Can you please approve this?

File: pilot/pkg/model/sidecar.go [hunk 2]
```diff
@@ -506,16 +505,12 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
...
 								sc := s.DeepCopy()
-								sc.Ports = []*Port{
-									{
-										ilw.IstioListener.Port.GetName(),
-										int(ilw.IstioListener.Port.GetNumber()),
-										protocol.Instance(ilw.IstioListener.Port.GetProtocol()),
-									},
-								}
+								ports = append(ports, port)
+								sc.Ports = ports
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar.go]: This includes all the ports?

Comment (ramaraochavali) [reply pilot/pkg/model/sidecar.go]: no.just matching ports

Comment (ramaraochavali) [reply pilot/pkg/model/sidecar.go]: earlier also it used to the same  - but copied from listener, now it just copies from service

Comment (hzxuzhonghu) [reply pilot/pkg/model/sidecar.go]: IC, could you move L508 down into the if branch?

**Comment by ramaraochavali** [comment]
> Sorry, the eds part completely escaped my mind when I approved your earlier PR.

No problem at all.

**Pull request merged**